pytorch_lightning/profiler/pytorch/pytorch_kineto.py

```diff
@@ -0,0 +1,338 @@
+                It will generate a ``.json`` file which can be read by Chrome.
+
+            row_limit: Limit the number of rows in a table, ``-1`` is a special value that
+                removes the limit completely.
+
+            sort_by_key: Attribute used to sort entries. By default
+                they are printed in the same order as they were registered.
+                Valid keys include: ``cpu_time``, ``cuda_time``, ``cpu_time_total``,
+                ``cuda_time_total``, ``cpu_memory_usage``, ``cuda_memory_usage``,
+                ``self_cpu_memory_usage``, ``self_cuda_memory_usage``, ``count``.
+
+            record_functions: Set of profiled functions which will create a context manager on.
+                Any other will be pass through.
+
+            record_module_names: Whether to add module names while recording autograd operation.
+
+            profiler_kwargs: Keyword arguments for the PyTorch profiler. This depends on your PyTorch version
+
+        Raises:
+            MisconfigurationException:
+                If arg ``sort_by_key`` is not present in ``AVAILABLE_SORT_KEYS``.
+        """
+        super().__init__(dirpath=dirpath, filename=filename)
+
+        self._group_by_input_shapes = group_by_input_shapes and profiler_kwargs.get("record_shapes", False)
+        self._emit_nvtx = emit_nvtx
+        self._export_to_chrome = export_to_chrome
+        self._row_limit = row_limit
+        self._sort_by_key = sort_by_key or f"{'cuda' if profiler_kwargs.get('use_cuda', False) else 'cpu'}_time_total"
+        self._user_record_functions = record_functions or set()
+        self._record_functions_start = self._user_record_functions | self.START_RECORD_FUNCTIONS
+        self._record_functions = self._user_record_functions | self.RECORD_FUNCTIONS
+        self._record_module_names = record_module_names
+        self._profiler_kwargs = profiler_kwargs
+
+        self.profiler: Optional[_PROFILER] = None
+        self.function_events: Optional["EventList"] = None
+        self._lightning_module: Optional["LightningModule"] = None  # set by ProfilerConnector
+        self._register: Optional[RegisterRecordFunction] = None
+        self._parent_profiler: Optional[_PROFILER] = None
+        self._recording_map: Dict[str, record_function] = {}
+
+        if _KINETO_AVAILABLE:
+            self._init_kineto_params(profiler_kwargs)
+
+        if self._sort_by_key not in self.AVAILABLE_SORT_KEYS:
+            raise MisconfigurationException(
+                f"Found sort_by_key: {self._sort_by_key}. Should be within {self.AVAILABLE_SORT_KEYS}. "
+            )
+
+        # calcuate the warmup and active step
+        self._action_step_num: Dict[str, int] = defaultdict(int)
+        self._wait_step: int = wait_step
+        self._warmup_step: int = warmup_step
+        self._active_step: int = active_step
+
+    def _init_kineto_params(self, profiler_kwargs: Any) -> None:
```

Comment:
Mind adding more docstring to the functions and comments?

Reply:
sure